soccer/gameplay/plays/defense/advanced_pileup.py

```diff
@@ -0,0 +1,91 @@
+import robocup
+import constants
+import play
+import skills
+import tactics
+import main
+import evaluation.position
+import math
+import random
+
+class AdvancedPileup(play.Play):
+
+    def __init__(self):
+        super().__init__(continuous=False)
+        ball = main.ball().pos
+        us = main.our_robots()
+        them = main.their_robots()
+        radius = constants.Ball.Radius
+        range = 20 * radius
+
+        c, pos = evaluation.position.ball_vicinity(us, them, range)
+        print(c, pos)
+
+        pos1 = None
+        pos2 = None
+        their_end = (length * 7)/8
+        our_end = length/8
+        left = base_width + (width/3)
+        right = base_width + (width*2)/3
+        dist = (length / 9)
+
+        #initial positioning
+        if (pos == -math.pi or pos == 0):
```

Comment:
Since these are `==`, it's very unlikely that this, or the first else if will be hit. It might be better to specify a range if you want these to be hit

Reply:
Wait, I just looked at this hard, see the code suggestion below and that should be enough to simplify a lot of this